cmake/CMakeshift/modules/FindHSL.cmake

```diff
@@ -0,0 +1,56 @@
+
+#.rst:
+# FindHSL
+# --------
+#
+# Find the HSL library.
```

Comment:
Nitpick: that apparently expands to "Find the Harwood Software Library library." Following the documentation, we probably should refer to it as the "HSL Mathematical Software Library". (Perhaps we should also provide a link to http://www.hsl.rl.ac.uk/index.html here.)